packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,297 @@
+
+### Shape
+
+The Avatar natively supports a circular and square shape. However, some use cases require a hexagon shape (e.g. to represent a bot). The hexagon will not be supported directly, but it is possible to style the Avatar with a hexagon background by using `tokens`:
+
+```jsx
+<Avatar
+  icon={<ChatBotIcon />}
+  display="icon"
+  tokens={{
+    width: 'calc(var(--avatar-height) * 1.125)',
+    background: `url('${hexagonSvg}') 0px/contain no-repeat`,
+    borderRadius: '0',
+  }}
+/>
+```
+
+Where hexagonSvg is defined as:
+
+```js
+hexagonSvg =
+  'data:image/svg+xml;utf8,' +
+  '<svg width="36" height="32" viewBox="0 0 36 32" fill="none" xmlns="http://www.w3.org/2000/svg">' +
+  '<path fill="rgb(232,232,232)" d="M0.407926 17.528C-0.135976 16.5859 -0.135975 15.4141 0.407926 14.472' +
+  'L7.91541 1.46793C8.44076 0.557947 9.39444 0 10.4245 0H25.5755C26.6056 0 27.5592 0.557951 28.0846 1.46793' +
+  'L35.5921 14.472C36.136 15.4141 36.136 16.5859 35.5921 17.528L28.0846 30.5321' +
+  'C27.5592 31.4421 26.6056 32 25.5755 32H10.4245C9.39443 32 8.44076 31.4421 7.91541 30.5321L0.407926 17.528Z"/>' +
+  '</svg>';
+```
+
+## API
+
+From [Avatar.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/src/components/Avatar/Avatar.types.tsx):
+
+### Slots
+
+- `as` - The root element of the Avatar.
+- `image` - The Avatar's image, if available.
+- `label` - The text shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
+- `icon` - Icon displayed when there's no image or intials available.
+- `badge` - Badge to show the avatar's status.
+
+### Props
+
+```ts
+export interface AvatarProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /** The Avatar's image. */
+  image?: ShorthandValue<{}>;
+
+  /** The label shown when there's no image. Defaults to the initials derived from `name` using `getInitials`. */
+  label?: ShorthandValue<{}>;
+
+  /** Icon displayed when there's no image or intials available, or if `display="icon"`. */
+  icon?: ShorthandValue<{}>;
+
+  /**
+   * Optional activity indicator
+   * * active: the avatar will be decorated according to activeDisplay
+   * * inactive: the avatar will be reduced in size and partially transparent
+   * * unset: normal display
+   *
+   * @defaultvalue unset
+   */
+  active?: 'active' | 'inactive' | 'unset';
+
+  /**
+   * The type of visual treatment to use when `active="active"`
+   * @defaultvalue ring
+   */
+  activeDisplay?: 'ring' | 'shadow' | 'glow' | 'ring-shadow' | 'ring-glow';
+
+  /**
+   * The color when displaying either an icon or initials.
+   * * neutral (default): gray
+   * * brand: color from the brand palette
+   * * colorful: pick a color from a set of pre-defined colors, based on a hash of the name prop. If another method is
+   *     needed to assign colors to avatars, use the colorIndex prop.
+   *
+   * @defaultvalue neutral
+   */
+  colorType?: 'neutral' | 'brand' | 'colorful';
```

Comment:
I'm open to naming feedback for `colorType` and the enum values.

Reply:
@behowell What about `palette`?

```
palette?: 'grayscale' | 'brand' | 'colorful';
```

Reply:
This is an avatar-specific concept, so I'm not sure if `palette` would be too generic and/or get confused with something related to theming? I originally had `colorScheme` but changed it from that because it was confusing with the theming concept of the same name. @xugao / @khmakoto do you have thoughts about `palette`?

Other options might be `colorVariant` or `colorSet`.

Also, re: `grayscale` vs. `neutral`. I'd have a _slight_ preference to go with `neutral` because that's the word that design uses, but I could easily be convinced to use `grayscale` if anyone feels strongly. In practice, users won't need to write that value anyways because it's the default.

Reply:
I like `colorVariant` the most since we "variant" is a term we are using in the converged world to define a particular look and feel.

Reply:
Ok I'll go with `colorVariant`. Thanks!